app/src/main/java/com/architectcoders/aacboard/ui/fragments/viewmodel/EditDashBoardViewModel.kt

```diff
@@ -0,0 +1,47 @@
+package com.architectcoders.aacboard.ui.fragments.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.architectcoders.aacboard.domain.data.dashboard.DashboardWithCells
+import com.architectcoders.aacboard.domain.use_case.dashboard.delete.DeleteDashboardUseCase
+import com.architectcoders.aacboard.domain.use_case.dashboard.get.GetDashboardUseCase
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class EditDashBoardViewModel(
+    private val dashBoardId: Int,
+    private val getDashboardUseCase: GetDashboardUseCase,
+    private val deleteDashboardUseCase: DeleteDashboardUseCase,
+) : ViewModel() {
+
+    private val _state = MutableStateFlow(EditDashBoardUiState())
+    val state: StateFlow<EditDashBoardUiState> get() = _state.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            _state.update { _state.value.copy(isLoading = true) }
+            getDashboardUseCase.invoke(dashBoardId).collect() { dashBoarWithCells ->
```

Comment:
(tip) when overriding the invoke operator there is no need to call the invoke method explicitly 
```suggestion
            getDashboardUseCase(dashBoardId).collect() { dashBoarWithCells ->
```

Reply:
Changed!